src/main/java/com/google/api/codegen/gapic/GapicProviderFactory.java

```diff
@@ -0,0 +1,202 @@
+      GapicProvider<? extends Object> provider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new ClientConfigGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new ClientConfigSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("json.snip"))
+          .build();
+      return Arrays.<GapicProvider<? extends Object>>asList(provider);
+
+    } else if (id.equals("csharp")) {
+      GapicProvider<? extends Object> provider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new CSharpGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new CSharpSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("wrapper.snip"))
+          .build();
+      return Arrays.<GapicProvider<? extends Object>>asList(provider);
+
+    } else if (id.equals("go")) {
+      GapicProvider<? extends Object> provider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new GoGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new GoSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("main.snip", "example.snip", "doc.snip"))
+          .build();
+      return Arrays.<GapicProvider<? extends Object>>asList(provider);
+
+    } else if (id.equals("java")) {
+      GapicProvider<? extends Object> mainProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new JavaGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new JavaSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("main.snip", "settings.snip"))
+          .build();
+      GapicProvider<? extends Object> packageInfoProvider = CommonGapicProvider.<Iterable<Interface>>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceListView())
+          .setContext(new JavaGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new JavaIterableSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("package-info.snip"))
+          .build();
+
+      return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, packageInfoProvider);
+
+    } else if (id.equals("nodejs")) {
+      GapicProvider<? extends Object> mainProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new NodeJSGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new NodeJSSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("main.snip"))
+          .build();
+      GapicProvider<? extends Object> clientConfigProvider = CommonGapicProvider.<Interface>newBuilder()
```

Comment:
I figured that since you are going to parameterize this specific class after this PR is submitted, I would leave that kind refactoring to you as a follow up step. Also, I kind of liked the simplicity and readability of this direct approach.